### ISSUE TITLE
[feature] Warn if no aliases.xml with devScenarion in tests

### ISSUE THREAD

**Issue by janhancic** [BladeRunnerJS/brjs#1201]
BRJS could output a warning to the console if you run tests that don't have an `aliases.xml` file that with `useScenario="dev"`.

Not having this is almost always a bug, and has on several occasions cost us quite some time.

**Comment by dchambers** [comment]
Was it the case that there was no `aliases.xml` file at all when this happened?

**Comment by janhancic** [comment]
Yes. But I suspect you would have the same issues if you had the file but didn't have \useScenario="dev"`.

In case it's not clear, the issues are that you don't get stub implementations of the services/aliases, you can't get templates from the HTML service, etc.

**Comment by dchambers** [comment]
@janhancic, we did consider making `useScenario="dev"` the default for tests, but didn't for reasons I can no longer remember. Even now, that may still be a better approach.

As for warnings, it wouldn't make sense to emit warnings if `aliases.xml` file existed (hence the question), since otherwise you effectively make `useScenario` a mandatory, rather than an optional, attribute.

**Comment by janhancic** [comment]
That is an issue to consider yes. But like I said, almost always you want the dev scenario in tests. I can't think of use case where you wouldn't want it. We don't have a single test folder that wouldn't be dev scenario.

So I think there should be a warning. If you really, really didn't want dev scenario, then you can just set `useScenario` to whatever the default one is (can't remember what it is). So you're making an explicit statement: "I know what I'm doing, don't bug me please".

Does that make sense?

**Comment by dchambers** [comment]
> But like I said, almost always you want the dev scenario in tests. I can't think of use case where you wouldn't want it. We don't have a single test folder that wouldn't be dev scenario.

That's a pretty strong case for making that the default in that case then. The only reasons not to are:

  1. It may confuse people that there is a different default scenario for tests.
  1. The `dev` scenario is not currently a _special_ scenario, and is just the convention we settled on, whereas it would become _special_.

> So I think there should be a warning. If you really, really didn't want dev scenario, then you can just set useScenario to whatever the default one is (can't remember what it is). So you're making an explicit statement: "I know what I'm doing, don't bug me please".

> Does that make sense?

I don't think having defaults that you get pestered for using are a good idea. I also don't think that having defaults that are almost always wrong is a good idea either, so I think one of my suggested tweaks to your proposal would be preferable.

@andyberry88 / @james-shaw-turner: Do you have an opinion on this?
